bundles/python/python_lexer.moon

```diff
@@ -0,0 +1,99 @@
+    "and", "as", "assert", "break", "class", "continue", "def", "del", "elif",
+    "else", "except", "exec", "finally", "for", "from", "global", "if", "import",
+    "in", "is", "lambda", "not", "or", "pass", "print", "raise", "return", "try",
+    "while", "with", "yield"
+  }
+
+  functions = c 'function', -B'.' * word {
+    "abs", "all", "any", "apply", "basestring", "bin",
+    "bool", "buffer", "bytearray", "bytes", "callable", "chr", "classmethod", "cmp",
+    "coerce", "compile", "complex", "copyright", "credits", "delattr", "dict",
+    "dir", "divmod", "enumerate", "eval", "execfile", "exit", "file", "filter",
+    "float", "format", "frozenset", "getattr", "globals", "hasattr", "hash", "help",
+    "hex", "id", "input", "intern", "int", "isinstance", "issubclass", "iter",
+    "len", "license", "list", "locals", "long", "map", "max", "memoryview", "min",
+    "next", "object", "oct", "open", "ord", "pow", "property", "quit",
+    "range", "raw_input", "reduce", "reload", "repr", "reversed", "round", "setattr",
+    "set", "slice", "sorted", "staticmethod", "str", "sum", "super", "tuple",
+    "type", "unichr", "unicode", "vars", "xrange", "zip"
+    }
+
+  constant = c 'constant', word {
+    'ArithmeticError', 'AssertionError', 'AttributeError', 'BaseException',
+    'BufferError', 'BytesWarning',
+    'DeprecationWarning', 'EOFError', 'Ellipsis', 'EnvironmentError', 'Exception',
+    'False', 'FloatingPointError', 'FutureWarning', 'GeneratorExit', 'IOError',
+    'ImportError', 'ImportWarning', 'IndentationError', 'IndexError', 'KeyError',
+    'KeyboardInterrupt', 'LookupError', 'MemoryError', 'NameError', 'None',
+    'NotImplementedError', 'NotImplemented', 'OSError', 'OverflowError',
+    'PendingDeprecationWarning', 'ReferenceError', 'RuntimeError',
+    'RuntimeWarning', 'StandardError', 'StopIteration', 'SyntaxError',
+    'SyntaxWarning', 'SystemError', 'SystemExit', 'TabError', 'True', 'TypeError',
+    'UnboundLocalError', 'UnicodeDecodeError', 'UnicodeEncodeError',
+    'UnicodeError', 'UnicodeTranslateError', 'UnicodeWarning', 'UserWarning',
+    'ValueError', 'Warning', 'ZeroDivisionError'
+  }
+
+  comment = c 'comment', P'#' * scan_until(eol)
+  operator = c 'operator', S'+-*/%~&^=<>;:,.(){}[]|'
+
+  name = (alpha + '_')^1 * (alpha + digit + S'_')^0
+
+  identifier = c 'identifier', name
+  fdecl = c('keyword', 'def') * c('whitespace', space^1) * c('fdecl', name)
+  classdef = c('keyword', 'class') * c('whitespace', space^1) * c('type_def', name)
+
+  hexadecimal =  P'0' * S'xX' * xdigit^1
+  octal = P'0' * S'oO'^-1 * R'07'^1
+  binary = P'0' * S'bB' * R'01'^1
+
+  digit_run = digit^1
+  float = digit_run * '.' * digit_run
+  integer = digit_run
+
+  basic_number = c 'number', any { hexadecimal, octal, binary, float, integer }
+
+  long_integer = c('number', digit_run) * c('operator', S'lL')
+  exponent_float = c('number', any { float, integer }) * c('operator', S'eE') * c('number', S'+-'^-1 * integer)
```

Comment:
Agree. I like the way this looks - it makes it harder to miss the important information. I also do this for the prefix in raw/encoded strings and the long number suffix(e.g. `r"abc"`, `123L`).

Do you think `special` makes more sense here?

LuaJIT's `ULL` is similar and both Rust and Nim allow suffixes like `123i32`  or `123'i32` to integers. Would be nice to style these literal modifiers differently.